### ISSUE TITLE
Okonomiyaki fails on existing egg with bad dependency specification

### ISSUE THREAD

**Issue by sjagoe** [enthought/okonomiyaki#89]
Okonomiyaki fails to parse egg metadata for `enthought/commercial/rh5-64/numexpr-2.2.2-3.egg`

```python
Traceback (most recent call last):
  File "okonomiyaki-test.py", line 130, in <module>
    main(import_dir, None)
  File "okonomiyaki-test.py", line 124, in main
    EggMetadata.from_egg(path)
  File "okonomiyaki/file_formats/_egg_info.py", line 627, in from_egg
    spec_depend = LegacySpecDepend.from_egg(path_or_file)
  File "okonomiyaki/file_formats/_egg_info.py", line 387, in from_egg
    return cls.from_string(spec_depend_string)
  File "okonomiyaki/file_formats/_egg_info.py", line 392, in from_string
    return cls._from_data(data, epd_platform)
  File "okonomiyaki/file_formats/_egg_info.py", line 372, in _from_data
    for s in args.get(_TAG_PACKAGES, [])
  File "okonomiyaki/file_formats/_egg_info.py", line 182, in from_spec_string
    raise InvalidRequirementString(name)
okonomiyaki.errors.InvalidRequirementString: Invalid requirement string 'numpy-1.8.0'
```

**Comment by cournape** [comment]
Ok, that one is valid: the metadata are indeed wrong.

That's a good candidate for the black list we talked about.

**Comment by cournape** [comment]
Fixed since `0.10.0` (in 3b3697f0b8585b3ba78fd751752a3245c537b2c4)
